main/lsp/requests/prepare_rename.cc

```diff
@@ -20,6 +20,41 @@ variant<JSONNullObject, unique_ptr<PrepareRenameResult>> getPrepareRenameResult(
     result->placeholder = name;
     return result;
 }
+
+variant<JSONNullObject, unique_ptr<PrepareRenameResult>>
+getPrepareRenameResultForSend(const core::GlobalState &gs, const core::lsp::SendResponse *sendResp) {
+    // The send expression is of the form <receiver>.<method><args>
+    // We want to return the range and placeholder for the method part of the expression, because this is what the
+    // editor will highlight in the rename UI.
+
+    // TODO(soam): handle dispatchResult->secondary
+    auto method = sendResp->dispatchResult->main.method;
+    if (!method.exists()) {
+        return JSONNullObject();
+    }
+    auto methodName = sendResp->dispatchResult->main.method.data(gs)->name.show(gs);
+    auto expr = sendResp->termLoc.source(gs);
+    // find the end of the receiver, find the dot, and then the next non-whitespace char
+    string::size_type receiverOffset = sendResp->receiverLoc.endPos() - sendResp->termLoc.beginPos();
+    if (receiverOffset != 0) {
+        receiverOffset = expr.find_first_of(".", receiverOffset) + 1;
+        receiverOffset = expr.find_first_not_of(" \t", receiverOffset);
+    }
+    auto offsets = sendResp->termLoc.offsets();
+    offsets.beginLoc += receiverOffset;
+    offsets.endLoc = offsets.beginLoc + methodName.length();
```

Comment:
Is this always the case, btw? (Could the source text refer to a shorter name somehow...? Should we verify that the source text at `offsets` contains `methodName`?)

I don't know the answers to these questions, so maybe we'll find out in testing. :)

Reply:
Hmm, interesting. I'll add a check. I can think of `new` vs `initialize`, but we don't want to rename those anyway... which I also need to add a check for.

(But also today I learned about `alias_method`, I'm not sure how Sorbet handles it maybe that's relevant here.)